### PR TITLE
doc/cephfs: add notes on application best practices

### PR DESCRIPTION
**Pull request by jcsp**
I was tidying up my home directory, and found this orphan .rst file that I must have written on a train at some point!

Putting it here to see if folks think it would be a useful addition.  It covers some FAQs, at the cost of perhaps scaring people a bit?

### CODE REVIEW

File: doc/cephfs/app-best-practices.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+number of files and then expect equivalent performance when you move
+to a much larger number of files.
+
+Do you need a filesystem?
+-------------------------
+
+Remember that Ceph also includes an object storage interface.  If your
+application needs to store huge flat collections of files where you just
+read and write whole files at once, then you might well be better off
+using :doc:`/radosgw/`
```
Comment (LenzGr) [inline doc/cephfs/app-best-practices.rst]: The docs rendering failed with `/home/jenkins-build/build/workspace/ceph-pr-render-docs/doc/cephfs/app-best-practices.rst:76:unknown document: /radosgw/`  - that reference should probably be replaced with a reference to `object-gateway`?
```
using :doc:`object_gateway`
```

File: doc/cephfs/app-best-practices.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+the inode after looking up the path).
+
+Working set size
+----------------
+
+The MDS acts as a cache for the metadata stored in RADOS.  Metadata
+performance is very different for workloads whose metadata fits within
+that cache.
+
+If your workload has more files than you ``mds_cache_size`` setting, then
```
Comment (gregsfortytwo) [inline doc/cephfs/app-best-practices.rst]: This config option is (un)fortunately out of date now. :)

**Comment by jcsp** [comment]
jenkins render docs

**Comment by jcsp** [comment]
Thanks for reviews, I've updated this for both comments.

File: doc/cephfs/app-best-practices.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+----------------------
+
+When you run "ls -l" (or use ls with coloring enabled), the ``ls`` program
+is first doing a directory listing, and then calling ``stat`` on every
+file in the directory.
+
+This is usually far in excess of what an application really needs, and
+it can be slow for large directories.  If you don't really need all
+this metadata for each file, then use a plain ``ls`` with coloring
+disabled.
```
Comment (batrick) [inline doc/cephfs/app-best-practices.rst]: I'm not sure this the case for ls with coloring. `ls` gets enough information from `getdents` to figure out the types of files (`d_type`) for coloring. In my own test just now, it only does `getdents`.

File: doc/cephfs/app-best-practices.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+ls on files being extended
```
Comment (batrick) [inline doc/cephfs/app-best-practices.rst]: I'd go ahead and extend this to `ls(1)/stat(1)` to be ultra-clear.

Comment (jcsp) [reply doc/cephfs/app-best-practices.rst]: Done

**Pull request merged**